mne/preprocessing/maxwell.py

```diff
@@ -642,3 +685,116 @@ def _overlap_projector(data_int, data_res, corr):
+        ch_ind = info['ch_names'].index(cal_chan['ch_name'])
+
+        # Loop over vectors in X, Y, and Z directions
+        for ang_i, (i1, i2) in enumerate(zip(range(3, 10, 3),
+                                             range(6, 13, 3))):
+            v1 = cal_chan['loc'][i1:i2]
+            v2 = info['chs'][ch_ind]['loc'][i1:i2].astype(float)
+
+            cos_ang = np.dot(v1, v2) / (np.linalg.norm(v1) *
+                                        np.linalg.norm(v2))
+            ang_shift[ch_ind, ang_i] = cos_ang
+
+        # Adjust channel orientation with those from fine calibration
+        info['chs'][ch_ind]['loc'] = cal_chan['loc']
+
+    # Deal with numerical precision giving values slightly more than 1.
+    np.clip(ang_shift, -1., 1., ang_shift)
+    ang_shift = np.rad2deg(np.arccos(ang_shift))  # Convert to degrees
+
+    # Log quantification of sensor changes
+    logger.info('Fine calibration adjusted coil positions by (mean ± std in '
+                'degrees): %0.1f ± %0.1f (max: %0.1f)' %
+                (np.mean(ang_shift), np.std(ang_shift),
+                 np.max(np.abs(ang_shift))))
+
+    return info
+
+
+def _sss_basis_point(origin, info, int_order, ext_order, imbalance,
+                     head_frame=True):
+    """Compute multipolar moments for point-like magnetometers (in fine cal)"""
+
+    # Permeability of free space for scaling
+    u0 = 4 * np.pi * 1e-7
+
+    # Construct 'coils' with r, weights, normal vecs, # integration pts, and
+    # channel type.
+    if imbalance.ndim == 1:
+        imbalance = imbalance[:, np.newaxis]
+    if imbalance.shape[1] not in [1, 3]:
+        raise ValueError('Must have 1 (x) or 3 (x, y, z) point-like ' +
+                         'magnetometers. Currently have %i' %
+                         imbalance.shape[1])
+
+    # Initialize internal multipolar moment space for point-like magnetometers
+    S_in_all = np.zeros((len(info['chs']), get_num_moments(int_order, 0)))
+    S_out_all = np.zeros((len(info['chs']), get_num_moments(0, ext_order)))
+
+    # Coil_type values for x, y, z point magnetometers
+    pt_types = [FIFF.FIFFV_COIL_POINT_MAGNETOMETER_X,
+                FIFF.FIFFV_COIL_POINT_MAGNETOMETER_Y,
+                FIFF.FIFFV_COIL_POINT_MAGNETOMETER]
+
+    # Loop over all coordinate directions desired and create point mags
+    for dir_ind in range(imbalance.shape[1]):
+
+        temp_info = deepcopy(info)
+        for ch in temp_info['chs']:
+            ch['coil_type'] = pt_types[dir_ind]
+        coils_add, _, _, _ = _prep_meg_channels(temp_info, accurate=True,
+                                                elekta_defs=True,
+                                                head_frame=head_frame)
+
+        S_in_add, S_out_add = _sss_basis(origin, coils_add, int_order,
+                                         ext_order)
+        # Scale spaces by gradiometer imbalance
+        S_in_all += S_in_add * imbalance[:, dir_ind][:, np.newaxis]
```

Comment:
`... * imbalance[:, [dir_ind]]` is more compact if `imbalance` is small the array copy (instead of view) it forces might be worth the smaller code